libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -545,12 +455,14 @@ - (void)syncUpOneEntry:(SFSyncState*)sync
     }
     
     NSString* idStr = [(NSNumber*) recordIds[i] stringValue];
-    NSMutableDictionary* record = [[self.store retrieveEntries:@[idStr] fromSoup:soupName][0] mutableCopy];
-    
+    NSMutableDictionary* record = [[target getFromLocalStore:self soupName:soupName storeId:idStr] mutableCopy];
+
+    LogSyncDebug(@"syncUpOneRecord:%@", record);
+
     // Do we need to do a create, update or delete
-    BOOL locallyCreated = [record[kSyncManagerLocallyCreated] boolValue];
-    BOOL locallyUpdated = [record[kSyncManagerLocallyUpdated] boolValue];
-    BOOL locallyDeleted = [record[kSyncManagerLocallyDeleted] boolValue];
+    BOOL locallyCreated = [target isLocallyCreated:record];
```

Comment:
One is free to flag locally created/updated/deleted record in any way one wants.